pages/exercises/[lessonSlug].tsx

```diff
@@ -208,24 +248,39 @@ const ExerciseList = ({
         />
       </div>
       <div className="d-flex flex-column flex-md-row justify-content-between align-items-center">
-        <h1 className="my-2 my-md-5 fs-2">{lessonTitle}</h1>
-        <div
-          className={`mb-3 mb-md-0 d-flex d-md-block ${styles.exerciseList__solveExercisesButtonContainer}`}
-        >
-          <NewButton
-            className="flex-grow-1"
-            onClick={() => setExerciseIndex(0)}
-          >
-            SOLVE EXERCISES
-          </NewButton>
+        <div className="my-2 my-md-5">
+          <h1 className="fs-2">{lessonTitle}</h1>
+          <label className="d-inline-flex align-items-center">
+            <input
+              className="form-check-input m-0 me-3"
+              type="checkbox"
+              style={{ width: 30, height: 30 }}
+              checked={hideAnswered}
+              onChange={() => setHideAnswered(!hideAnswered)}
+            />
+            <span>Show incomplete exercises only</span>
+          </label>
         </div>
+        {exercises.length > 0 && (
+          <div
+            className={`mb-3 mb-md-0 d-flex d-md-block ${styles.exerciseList__solveExercisesButtonContainer}`}
+          >
+            <NewButton className="flex-grow-1" onClick={onClickSolveExercises}>
+              SOLVE EXERCISES
+            </NewButton>
+          </div>
+        )}
```

Comment:
The SOLVE EXERCISES button only shows if there are exercises left. So if the user completes all the exercises and then checks the exercise filtering checkbox, there will be no exercises left, so this button is not shown.

Maybe we can also add a message in the empty exercise list body like "Congratulations! You finished all the exercises!" and then maybe we can add a button that says "Go to the next lesson" or something similar.